src/drivers/camera_trigger/camera_trigger.cpp

```diff
@@ -357,33 +365,33 @@ CameraTrigger::update_intervalometer()
 void
 CameraTrigger::update_distance()
 {
-	if (_turning_on) {
+	if (_turning_on || !_trigger_enabled || _trigger_paused) {
 		return;
 	}
 
-	if (_trigger_enabled) {
-		vehicle_local_position_s local{};
-		_lpos_sub.copy(&local);
+	vehicle_local_position_s local{};
+	_lpos_sub.copy(&local);
 
-		if (local.xy_valid) {
+	if (local.xy_valid) {
 
-			// Initialize position if not done yet
-			matrix::Vector2f current_position(local.x, local.y);
+		// Initialize position if not done yet
+		matrix::Vector2f current_position(local.x, local.y);
 
-			if (!_valid_position) {
-				// First time valid position, take first shot
-				_last_shoot_position = current_position;
-				_valid_position = local.xy_valid;
-				shoot_once();
-			}
+		if (!_valid_position) {
+			// First time valid position, take first shot
+			_last_shoot_position = current_position;
+			_valid_position = local.xy_valid;
 
-			// Check that distance threshold is exceeded
-			if (matrix::Vector2f(_last_shoot_position - current_position).length() >= _distance) {
+			if (!_one_shot) {
```

Comment:
It is orthogonal because it tells the system to take a single picture and disengage right after. If you wouldn't architect it this way you would need to schedule a HRT call to disengage the trigger right after, opening the door to horrible race conditions.